core/src/main/java/com/kin/ecosystem/core/data/auth/AuthRepository.java

```diff
@@ -33,6 +33,8 @@ private AuthRepository(@NonNull AuthDataSource.Local local,
 		@NonNull AuthDataSource.Remote remote) {
 		this.localData = local;
 		this.remoteData = remote;
+		this.cachedSignInData = local.getSignInData();
```

Comment:
should happen but just to be on the safe side also check cachedSignInData != null in updateWalletAddress()

Reply:
there is no need, you already logged in and just updating wallet address.